### PR TITLE
Sidebar: Removed all tracking events from non unified-nav sidebar menu item clicks.

### PR DESCRIPTION
**Pull request by rcrdortiz**
I've removed all the calypso_mysites_sidebar_*_click events generated by clicking on menu items of the non-unified-nav sidebar.
I've also fixed a small issue, when the user clicked on the 'Domains' menu, we were triggering a nudge cta click event, even though the menu item clearly isn't a nudge. Now we aren't sending that event.

#### Changes proposed in this Pull Request

* Removed menu item click events in the non-unified-nav sidebar.
* The domains menu doesn't trigger a nudge cta click event.

Related to #50847

##### Test that no tracking events are being sent for the menu item clicks.
1. Open the Calypso live image in your browser and then.
2. Open Chrome's inspector and add the following line to the console: localStorage.debug='calypso:analytics'
3. Make sure that Chrome's console is logging all levels and not just errors and warnings.
4. Reload the page.
5. Now add ?disable-nav-unification to the URL and you should be in the non-nav-unification version.
6. Click on a Calypso link, in Chrome's console you shouldn't see any menu item click tracking being triggered.

##### Test that the domain menu item isn't triggering a nudge click tracking event.
1. Follow the previous steps up until step 5.
2. Click on the domain menu item.
3. Check Chrome's console and make sure that no nudge click event (calypso_domain_credit_reminder_click) is being tracked.
4. [Optional] Check that the domain nudge triggers a calypso_domain_credit_reminder_click tracking event.

#### Test evidence
https://user-images.githubusercontent.com/1989914/146971524-a785f14f-fb26-45e5-a264-605bbe9e9ea3.mp4

### CODE REVIEW

File: client/my-sites/sidebar/index.jsx [hunk 1]
```diff
@@ -769,25 +724,6 @@ export class MySitesSidebar extends Component {
...
-		this.props.recordTracksEvent(
-			'calypso_mysites_sidebar_' + menuItemName.replace( /-/g, '_' ) + '_clicked'
-		);
-	};
-
-	trackPlanClick = () => {
-		this.trackMenuItemClick( 'plan' );
-		this.props.recordTracksEvent( 'calypso_upgrade_nudge_cta_click', {
-			cta_name: 'sidebar_upgrade_default',
-		} );
```
Comment (rcrdortiz) [inline client/my-sites/sidebar/index.jsx:782]: This is where we are triggering the nudge cta click event. This function is only called by the domains menu item.

**Comment by mmtr** [comment]
> I've removed all the calypso_mysites_sidebar_*_click events generated by clicking on menu items of the non-unified-nav sidebar.

I guess we should also deprecate them in the Tracks Events directory (see PCYsg-ktp-p2#what-do-the-colors-mean)

File: client/my-sites/sidebar/index.jsx [hunk 2]
```diff
@@ -646,9 +604,6 @@ export class MySitesSidebar extends Component {
 	}
 
 	trackWooCommerceNavItemClick = ( menuItemName, experience, plan ) => {
```
Comment (mmtr) [inline client/my-sites/sidebar/index.jsx]: I think we can remove this whole function.

**Review by mmtr — APPROVED** [review]
Looks great! We'll be eventually removing the old sidebar completely as part of https://github.com/Automattic/wp-calypso/issues/47051, but this is a good first step!

File: client/my-sites/sidebar/index.jsx (hunk 2 above)
Comment (rcrdortiz) [reply client/my-sites/sidebar/index.jsx]: I wasn't sure if the calypso_woocommerce_nav_item_click was being used and that's why I didn't remove it initially.  I'll delete the function.

Done.

Comment (rcrdortiz) [reply client/my-sites/sidebar/index.jsx]: Done.

**Review by mmtr — APPROVED** [review]

**Comment by rcrdortiz** [comment]
> > I've removed all the calypso_mysites_sidebar_*_click events generated by clicking on menu items of the non-unified-nav sidebar.
> 
> I guess we should also deprecate them in the Tracks Events directory (see PCYsg-ktp-p2#what-do-the-colors-mean)

Flagged all the calypso_mysites_sidebar_*_click events as deprecated in Tracks. You can use this query string for checking: ?_status=all&_name=calypso_mysites_sidebar

**Pull request merged**